crates/sparrow-scheduler/src/sink.rs

```diff
@@ -11,19 +11,69 @@ pub struct PipelineInput {
     input: usize,
 }
 
+#[derive(Debug, Default)]
+pub struct PipelineInputs(smallvec::SmallVec<[PipelineInput; 1]>);
+
+impl PipelineInputs {
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    pub fn add_consumer(&mut self, pipeline: Arc<dyn Pipeline>, input: usize) {
+        self.0.push(PipelineInput::new(pipeline, input));
```

Comment:
Added this -- can probably delete or hide `PipelineInput` but basically needed to be able to say "this pipeline has multiple consumers"